src/walletConnect/v1/saga.ts

```diff
@@ -299,6 +301,17 @@ function* listenForWalletConnectMessages(walletConnectChannel: EventChannel<Wall
   }
 }
 
+// Though the WC types say `icons` is string[], we've seen buggy clients with no `icons` property
+// so to avoid crashing the code depending on this, we fix it here
+// Note: this method mutates the session
+function applyIconFixIfNeeded(session: WalletConnectSessionRequest | WalletConnectSession) {
+  const { peerMeta } = 'peerMeta' in session ? session : session.params[0]
```

Comment:
is there any chance that `session.params[0]` is undefined? 🙈 it's probably not likely but if we default to an empty object then this function shouldn't ever fail? idk, maybe overkill

Reply:
Good question, I think it should fail if this happens.
We won't be able to do anything with that request and it should propagate the error.